test/edge-cases.js

```diff
@@ -14,10 +14,12 @@ describe( 'Edge cases', function() {
 
     pipeline
       .on( 'error', done )
-      .on( 'finish', done )
+      .on( 'end', done )
       .on( 'readable', function() {
         var hash = this.read()
-        assert.equal( hash.toString( 'hex' ), '37a309ae59ad3a562a083eb90e32fa6757bdd41f' )
+        if( hash != null ) {
```

Comment:
So if the returned hash is always null, the test still passes??
Is this because the hash is null first time, but then set the second time, or something? Maybe it'd be worth checking that we see at least one non-null hash during the unit-test?